src/dbnode/persist/fs/write_large_tiles_test.go

```diff
@@ -0,0 +1,224 @@
+	xtime "github.com/m3db/m3/src/x/time"
+	"github.com/stretchr/testify/require"
+)
+
+type testLargeTileEntry struct {
+	testEntry
+	values []float64
+}
+
+func newTestLargeTilesWriter(
+	t *testing.T,
+	filePathPrefix string,
+	shard uint32,
+	timestamp time.Time,
+	nextVersion int,
+	plannedEntries uint,
+) LargeTilesWriter {
+	writer, err := NewLargeTilesWriter(
+		LargeTilesWriterOptions{
+			NamespaceID: testNs1ID,
+			ShardID:     shard,
+			BlockStart:  timestamp,
+			BlockSize:   testBlockSize,
+
+			VolumeIndex:         nextVersion,
+			PlannedRecordsCount: plannedEntries,
+			Options: testDefaultOpts.
+				SetFilePathPrefix(filePathPrefix).
+				SetWriterBufferSize(testWriterBufferSize),
+		},
+	)
+	require.NoError(t, err)
+	return writer
+}
+
+func TestIdsMustBeSorted(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"bar", nil, nil}, []float64{4.8, 5.2, 6}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	defer w.Close()
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.Error(t, err)
+	require.Equal(t, "ids must be written in lexicographic order, no duplicates, but got baz followed by bar",
+		err.Error())
+}
+
+func TestDoubleWritesAreNotAllowed(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"baz", nil, nil}, []float64{4.8, 5.2, 6}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	defer w.Close()
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.Error(t, err)
+	require.Equal(t, "ids must be written in lexicographic order, no duplicates, but got baz followed by baz",
+		err.Error())
+}
+
+func TestSimpleLargeTilesReadWrite(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"bar", nil, nil}, []float64{4.8, 5.2, 6}},
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"cat", nil, nil}, []float64{100000}},
+		{testEntry{"foo", nil, nil}, []float64{1, 2, 3}},
+		{testEntry{"foo+bar=baz,qux=qaz", map[string]string{
+			"bar": "baz",
+			"qux": "qaz",
+		}, nil}, []float64{7, 8, 9}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.NoError(t, err)
+	err = w.Close()
+	require.NoError(t, err)
+
+	expectEntries := make([]testEntry, 0, len(entries))
+	for _, e := range entries {
+		expectEntries = append(expectEntries, e.testEntry)
+	}
+
+	r := newTestReader(t, filePathPrefix)
+	readTestDataWithOrderOpt(t, r, 0, testWriterStart, expectEntries, true)
```

Comment:
I might be missing something, but is there any place where it checks the actual datapoint values written?

Reply:
It compares bytes passed: https://github.com/m3db/m3/blob/gg/large-tiles-writer-tests/src/dbnode/persist/fs/read_write_test.go#L241
When we encode datapoints into bytes, we store bytes representation with series and later compare it: https://github.com/m3db/m3/pull/2608/files#diff-d62e9a6b4cbf4aaa3f88808e12f7952dR215